docker-entrypoint.sh

```diff
@@ -74,4 +74,14 @@ fi
 
 # these 3 vars are used by `/go-server/server.sh`, so we export
 export GO_SERVER_SYSTEM_PROPERTIES="${GO_SERVER_SYSTEM_PROPERTIES}${GO_SERVER_SYSTEM_PROPERTIES:+ }-Dgo.console.stdout=true"
-try exec "$@"
+
+echo "Running custom scripts..."
+for f in /docker-entrypoint.d/*; do
+  case "$f" in
+    *.sh)     echo "$0: running $f"; . "$f" ;;
+    *)        echo "$0: ignoring $f" ;;
+  esac
+  echo
+done
```

Comment:
I'd rather that we don't `source` files. Sourcing implies that scripts must be written in shell. This will limit extensibility preventing executables written in other languages from being executed.

An alternative would be to do something like this (**untested code***):

```shell
for file in /docker-entrypoint.d/*.sh; do
  if [ -f "$file" ] && [ -x "$file" ]; then
    try "$file"
  else
    warn "Ignoring $file"
  fi
done
```

Reply:
exec / try is exactly what is best practice is due to process forking, exit codes and so on - i am just using what they all use. But feel free to explain why this is bad. 

Why should we not run ruby scripts if the interpreter is there? or zsh, bash or sh?

Reply:
> Why should we not run ruby scripts if the interpreter is there? or zsh, bash or sh?

Isn't that what @ketan is saying? Your code allows only shell scripts (well, scripts ending in .sh) and then *sources* them. Ketan is suggesting checking whether they're executable, and executing them - not sourcing them. So, that will allow ruby scripts, zsh, whatever. Unlike when using "source".

Reply:
Just checked https://github.com/docker-library/percona/blob/52abf70205354001f24432c0c677be4a0264ef47/5.7/docker-entrypoint.sh#L167 they using exactly the same implementation that i did - and i have seen that one all over the place.

I cannot really judge yours, its up to you. For those things i always pick the most common, most tested way, not the one i might think is more elegant or cleaner. Up to you, i am for sure fine if we do not run into issues.

And you are right, this is shell only, i obviously swapped that, sorry.

Reply:
@ketan - is it necessary to run the custom scripts for something like `docker run -it <image>:<tag> /bin/bash` ? That is, is running any script necessary if the go server is not going to be start when starting the container?

Reply:
@varshavaradarajan since docker-run is starting a new container, i doubt that is ever done with the gocd-server, but in terms of what the user wants here, you want to ensure that those scripts are run. Since the entrypoint runs those, and then executes CMD, run .. bash will execute this scripts, as this will setup a consistent gocd-server environment every single time.

Beside that, there is hardly any single case to do a oneshot `docker run -it gocd/gocd-server /bin/bash` except debugging the image creating in general.